In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import duckdb as db
import matplotlib.pyplot as plt
import scipy.stats as stats

In [3]:
df_dietary_1 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DR1IFF.XPT')
df_dietary_2 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DR2IFF.XPT')

In [66]:
df_total_nutrient_1 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DR1TOT.XPT')
df_total_nutrient_2 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DR2TOT.XPT')

/var/folders/jk/45pfmj415lb9l0__2lq4xmr40000gn/T/ipykernel_20430/2422103708.py:1: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df_total_nutrient_1 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DR1TOT.XPT')
/var/folders/jk/45pfmj415lb9l0__2lq4xmr40000gn/T/ipykernel_20430/2422103708.py:1: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df_total_nutrient_1 = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Doc

In [5]:
df_demo = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_DEMO.XPT')

In [103]:
df_BMX = pd.read_sas('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/P_BMX.XPT')

In [8]:
df_dietary_1

SEQN     WTDRD1PP      WTDR2DPP  DR1ILINE  DR1DRSTZ  DR1EXMER  \
0       109263.0  7619.483586  1.780807e+04      12.0       1.0      14.0   
1       109263.0  7619.483586  1.780807e+04      10.0       1.0      14.0   
2       109263.0  7619.483586  1.780807e+04       4.0       1.0      14.0   
3       109263.0  7619.483586  1.780807e+04       9.0       1.0      14.0   
4       109263.0  7619.483586  1.780807e+04      13.0       1.0      14.0   
...          ...          ...           ...       ...       ...       ...   
183905  124821.0  3043.724330  5.397605e-79       5.0       1.0      81.0   
183906  124821.0  3043.724330  5.397605e-79       4.0       1.0      81.0   
183907  124821.0  3043.724330  5.397605e-79       3.0       1.0      81.0   
183908  124821.0  3043.724330  5.397605e-79       2.0       1.0      81.0   
183909  124821.0  3043.724330  5.397605e-79       1.0       1.0      81.0   

        DRABF  DRDINT       DR1DBIH  DR1DAY  ...      DR1IM181      DR1IM201  \
0         2.0     2.0  4.000000e+00     6.0  ...  1.941000e+00  3.600000e-02   
1         2.0     2.0  4.000000e+00     6.0  ...  1.229000e+00  1.300000e-02   
2         2.0     2.0  4.000000e+00     6.0  ...  5.683000e+00  8.800000e-02   
3         2.0     2.0  4.000000e+00     6.0  ...  1.000000e-02  5.397605e-79   
4         2.0     2.0  4.000000e+00     6.0  ...  2.382000e+00  1.400000e-02   
...       ...     ...           ...     ...  ...           ...           ...   
183905    2.0     1.0  5.397605e-79     7.0  ...  5.397605e-79  5.397605e-79   
183906    2.0     1.0  5.397605e-79     7.0  ...  1.600000e+00  5.397605e-79   
183907    2.0     1.0  5.397605e-79     7.0  ...  3.900000e-01  4.000000e-03   
183908    2.0     1.0  5.397605e-79     7.0  ...  5.397605e-79  5.397605e-79   
183909    2.0     1.0  5.397605e-79     7.0  ...  2.018700e+01  4.820000e-01   

            DR1IM221      DR1IP182      DR1IP183      DR1IP184      DR1IP204  \
0       5.397605e-79  8.380000e-01  8.800000e-02  5.397605e-79  1.200000e-02   
1       5.397605e-79  5.200000e-01  1.300000e-02  5.397605e-79  5.397605e-79   
2       5.000000e-03  3.482000e+00  4.210000e-01  5.000000e-03  2.900000e-02   
3       5.397605e-79  6.900000e-02  1.500000e-02  5.397605e-79  5.397605e-79   
4       5.397605e-79  1.209000e+00  1.140000e-01  5.397605e-79  5.397605e-79   
...              ...           ...           ...           ...           ...   
183905  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
183906  2.000000e-03  2.632000e+00  2.010000e-01  5.397605e-79  1.000000e-03   
183907  5.397605e-79  6.530000e-01  7.900000e-02  5.397605e-79  9.000000e-03   
183908  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
183909  1.800000e-02  1.074500e+01  1.281000e+00  9.000000e-03  2.480000e-01   

            DR1IP205      DR1IP225      DR1IP226  
0       5.397605e-79  5.397605e-79  5.397605e-79  
1       5.397605e-79  5.397605e-79  5.397605e-79  
2       1.000000e-02  1.000000e-02  5.397605e-79  
3       5.397605e-79  5.397605e-79  5.397605e-79  
4       5.397605e-79  5.397605e-79  5.397605e-79  
...              ...           ...           ...  
183905  5.397605e-79  5.397605e-79  5.397605e-79  
183906  5.397605e-79  5.397605e-79  5.397605e-79  
183907  1.500000e-02  1.000000e-03  1.600000e-02  
183908  5.397605e-79  5.397605e-79  5.397605e-79  
183909  9.600000e-02  8.300000e-02  3.210000e-01  

[183910 rows x 84 columns]

In [116]:
q_1 = db.sql('select SEQN, DR1DAY, sum(DR1IKCAL) TOT1KCAL , sum(DR1IPROT) TOT1PROT, sum(DR1ICARB) TOT1CARB, sum(DR1ISUGR) TOT1SUGR, sum(DR1IFIBE) TOT1FIBE, sum(DR1ITFAT) TOT1FAT, sum(DR1IATOC) TOT1ATOC, sum(DR1ISODI) TOT1SODI from df_dietary_1 group by SEQN, DR1DAY').df()
q_1

SEQN  DR1DAY  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  \
0      118027.0     6.0     771.0     41.45     89.37     28.08       8.4   
1      118039.0     5.0     849.0     21.17     80.91     30.36       9.6   
2      118040.0     3.0    3605.0    119.29    470.53    225.58      15.5   
3      118044.0     1.0    1864.0     68.89    128.94     40.15      13.4   
4      118046.0     6.0    1536.0     46.06    162.79     62.28      13.0   
...         ...     ...       ...       ...       ...       ...       ...   
12627  109454.0     6.0     478.0     12.76     69.01     37.02       2.9   
12628  110214.0     7.0    3935.0    149.58    390.50    131.38      14.9   
12629  114946.0     6.0     760.0     18.07    131.07     81.21       7.9   
12630  115964.0     6.0    1264.0     35.36    163.69     81.67       9.4   
12631  117635.0     5.0    1027.0     30.75    148.87    142.92       0.1   

       TOT1FAT  TOT1ATOC  TOT1SODI  
0        29.39      2.12    1411.0  
1        23.75     10.46     817.0  
2       141.74     12.15    5932.0  
3       119.34     10.03    4153.0  
4        78.80      5.20    2521.0  
...        ...       ...       ...  
12627    17.45      2.47     655.0  
12628   201.07     11.70    8491.0  
12629    19.80      2.53     947.0  
12630    54.41      5.64    1337.0  
12631    37.51      1.36    1357.0  

[12632 rows x 10 columns]

In [118]:
q_2 = db.sql('select SEQN, DR2DAY, sum(DR2IKCAL) TOT2KCAL , sum(DR2IPROT) TOT2PROT, sum(DR2ICARB) TOT2CARB, sum(DR2ISUGR) TOT2SUGR, sum(DR2IFIBE) TOT2FIBE, sum(DR2ITFAT) TOT2FAT, sum(DR2IATOC) TOT2ATOC, sum(DR2ISODI) TOT2SODI from df_dietary_2 group by SEQN, DR2DAY').df()
q_2

SEQN  DR2DAY  TOT2KCAL  TOT2PROT  TOT2CARB  TOT2SUGR      TOT2FIBE  \
0      120015.0     5.0    1630.0     43.99    202.10    148.03  4.000000e+00   
1      120016.0     1.0    5356.0    163.59    669.16    279.90  2.700000e+01   
2      120018.0     5.0     795.0     28.74    127.62     73.32  3.238563e-78   
3      120023.0     7.0    1508.0     60.72    196.71    137.23  1.450000e+01   
4      120031.0     3.0    1772.0     75.87    263.14    101.87  1.870000e+01   
...         ...     ...       ...       ...       ...       ...           ...   
10825  116480.0     5.0    3038.0     88.16    298.17    111.48  2.920000e+01   
10826  117970.0     5.0    1334.0     54.76    183.72    100.86  1.020000e+01   
10827  117989.0     4.0    3227.0    118.51    436.37    208.98  3.200000e+01   
10828  111813.0     4.0    1556.0     59.67    222.01     65.47  1.850000e+01   
10829  114275.0     2.0     563.0     32.27     74.53     15.22  9.000000e+00   

       TOT2FAT  TOT2ATOC  TOT2SODI  
0        72.34      3.96    2146.0  
1       230.25     14.01    9819.0  
2        19.14     10.74     657.0  
3        56.35      3.64    1282.0  
4        48.95      5.28    3592.0  
...        ...       ...       ...  
10825   174.48     18.29    2938.0  
10826    44.07      3.38    1600.0  
10827   107.06     22.61    5196.0  
10828    49.40      8.40    2886.0  
10829    15.59      1.12     869.0  

[10830 rows x 10 columns]

In [119]:
merged_df = pd.merge(q_1, q_2, on='SEQN', how='outer')
#merged_df = pd.merge(q_1, df_demo, on='SEQN', how='left')
merged_df

SEQN  DR1DAY  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  \
0      118027.0     6.0     771.0     41.45     89.37     28.08       8.4   
1      118039.0     5.0     849.0     21.17     80.91     30.36       9.6   
2      118040.0     3.0    3605.0    119.29    470.53    225.58      15.5   
3      118044.0     1.0    1864.0     68.89    128.94     40.15      13.4   
4      118046.0     6.0    1536.0     46.06    162.79     62.28      13.0   
...         ...     ...       ...       ...       ...       ...       ...   
12629  114946.0     6.0     760.0     18.07    131.07     81.21       7.9   
12630  115964.0     6.0    1264.0     35.36    163.69     81.67       9.4   
12631  117635.0     5.0    1027.0     30.75    148.87    142.92       0.1   
12632  117012.0     NaN       NaN       NaN       NaN       NaN       NaN   
12633  110939.0     NaN       NaN       NaN       NaN       NaN       NaN   

       TOT1FAT  TOT1ATOC  TOT1SODI  DR2DAY  TOT2KCAL  TOT2PROT  TOT2CARB  \
0        29.39      2.12    1411.0     3.0    1132.0     50.45    135.95   
1        23.75     10.46     817.0     6.0    1673.0     62.02    137.97   
2       141.74     12.15    5932.0     1.0    1129.0     60.53     75.71   
3       119.34     10.03    4153.0     5.0     918.0     24.48    128.91   
4        78.80      5.20    2521.0     NaN       NaN       NaN       NaN   
...        ...       ...       ...     ...       ...       ...       ...   
12629    19.80      2.53     947.0     4.0     983.0     30.30    145.26   
12630    54.41      5.64    1337.0     NaN       NaN       NaN       NaN   
12631    37.51      1.36    1357.0     1.0    1696.0     97.39    190.46   
12632      NaN       NaN       NaN     2.0     794.0     36.61     79.97   
12633      NaN       NaN       NaN     1.0    2588.0     47.24    434.63   

       TOT2SUGR  TOT2FIBE  TOT2FAT  TOT2ATOC  TOT2SODI  
0         32.78       8.3    44.93      6.49    2009.0  
1         24.01       8.2    72.55      6.79    2972.0  
2         12.49       3.4    63.68      2.88    2359.0  
3         54.84       3.1    34.97      0.73    1135.0  
4           NaN       NaN      NaN       NaN       NaN  
...         ...       ...      ...       ...       ...  
12629     86.34      13.8    35.73     10.88    1300.0  
12630       NaN       NaN      NaN       NaN       NaN  
12631    121.68      10.9    61.90      4.20    2855.0  
12632     31.37       6.0    36.91      3.57    1005.0  
12633    264.81      18.3    61.45      4.64    3129.0  

[12634 rows x 19 columns]

In [120]:
merged_df.T

0          1          2          3          4          5      \
SEQN      118027.00  118039.00  118040.00  118044.00  118046.00  118064.00   
DR1DAY         6.00       5.00       3.00       1.00       6.00       6.00   
TOT1KCAL     771.00     849.00    3605.00    1864.00    1536.00    2144.00   
TOT1PROT      41.45      21.17     119.29      68.89      46.06      92.70   
TOT1CARB      89.37      80.91     470.53     128.94     162.79     230.25   
TOT1SUGR      28.08      30.36     225.58      40.15      62.28      60.93   
TOT1FIBE       8.40       9.60      15.50      13.40      13.00      11.80   
TOT1FAT       29.39      23.75     141.74     119.34      78.80      94.48   
TOT1ATOC       2.12      10.46      12.15      10.03       5.20       5.21   
TOT1SODI    1411.00     817.00    5932.00    4153.00    2521.00    5282.00   
DR2DAY         3.00       6.00       1.00       5.00        NaN       2.00   
TOT2KCAL    1132.00    1673.00    1129.00     918.00        NaN    3151.00   
TOT2PROT      50.45      62.02      60.53      24.48        NaN     158.54   
TOT2CARB     135.95     137.97      75.71     128.91        NaN     284.20   
TOT2SUGR      32.78      24.01      12.49      54.84        NaN      72.01   
TOT2FIBE       8.30       8.20       3.40       3.10        NaN      25.90   
TOT2FAT       44.93      72.55      63.68      34.97        NaN     154.04   
TOT2ATOC       6.49       6.79       2.88       0.73        NaN       6.86   
TOT2SODI    2009.00    2972.00    2359.00    1135.00        NaN    4558.00   

              6          7          8          9      ...         12624  \
SEQN      118066.00  118082.00  118087.00  118112.00  ...  1.178550e+05   
DR1DAY         6.00       5.00       1.00       6.00  ...  7.000000e+00   
TOT1KCAL    2370.00    2034.00    1525.00    4200.00  ...  6.480000e+02   
TOT1PROT      54.77      61.43      51.60     141.49  ...  1.368000e+01   
TOT1CARB     313.25     194.93     292.85     664.99  ...  7.200000e+01   
TOT1SUGR     166.00      31.52     144.95     204.35  ...  7.200000e+01   
TOT1FIBE      11.90      18.80      26.50      65.10  ...  4.318084e-78   
TOT1FAT      101.92     101.48      24.93     111.08  ...  3.408000e+01   
TOT1ATOC      11.64      15.00       4.27      14.51  ...  8.720000e+00   
TOT1SODI    3050.00    2385.00    1748.00    6831.00  ...  1.760000e+02   
DR2DAY         4.00        NaN       2.00       2.00  ...           NaN   
TOT2KCAL    1985.00        NaN     653.00    1336.00  ...           NaN   
TOT2PROT      37.65        NaN      18.36      39.27  ...           NaN   
TOT2CARB     245.86        NaN     127.77     122.32  ...           NaN   
TOT2SUGR      81.06        NaN      21.71       3.84  ...           NaN   
TOT2FIBE      11.20        NaN      12.10      15.40  ...           NaN   
TOT2FAT       97.96        NaN       8.63      76.01  ...           NaN   
TOT2ATOC       9.75        NaN       1.33      10.16  ...           NaN   
TOT2SODI    2985.00        NaN     828.00    1902.00  ...           NaN   

              12625      12626      12627      12628      12629      12630  \
SEQN      113616.00  110038.00  109454.00  110214.00  114946.00  115964.00   
DR1DAY         3.00       1.00       6.00       7.00       6.00       6.00   
TOT1KCAL    2476.00    1794.00     478.00    3935.00     760.00    1264.00   
TOT1PROT      83.00     133.25      12.76     149.58      18.07      35.36   
TOT1CARB     219.52     218.70      69.01     390.50     131.07     163.69   
TOT1SUGR      80.31     114.55      37.02     131.38      81.21      81.67   
TOT1FIBE      10.90      22.70       2.90      14.90       7.90       9.40   
TOT1FAT      132.33      46.16      17.45     201.07      19.80      54.41   
TOT1ATOC      10.95       5.90       2.47      11.70       2.53       5.64   
TOT1SODI    3774.00    5443.00     655.00    8491.00     947.00    1337.00   
DR2DAY         1.00        NaN       5.00       4.00       4.00        NaN   
TOT2KCAL    2609

### MANUAL SELECTION of respondents

In [55]:
merged_df.loc[merged_df['SEQN'] == 118030.0]

SEQN  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  TOT1FAT  \
0  118030.0    1179.0     33.77    129.31     76.44       8.7    60.66   

   TOT1ATOC  TOT1SODI  TOT2KCAL  TOT2PROT  TOT2CARB  TOT2SUGR  TOT2FIBE  \
0      5.51    1247.0     835.0     27.71     78.13     39.84       2.5   

   TOT2FAT  TOT2ATOC  TOT2SODI  
0    40.51      2.54    1588.0

In [63]:
merged_df.loc[merged_df['SEQN'] == 118030.0].T

0
SEQN      118030.00
TOT1KCAL    1179.00
TOT1PROT      33.77
TOT1CARB     129.31
TOT1SUGR      76.44
TOT1FIBE       8.70
TOT1FAT       60.66
TOT1ATOC       5.51
TOT1SODI    1247.00
TOT2KCAL     835.00
TOT2PROT      27.71
TOT2CARB      78.13
TOT2SUGR      39.84
TOT2FIBE       2.50
TOT2FAT       40.51
TOT2ATOC       2.54
TOT2SODI    1588.00

### RANDOM SELECTION of respondents

In [56]:
import random

In [57]:
merged_df['SEQN'].describe()

count     12634.000000
mean     117103.994697
std        4499.703167
min      109263.000000
25%      113222.250000
50%      117147.500000
75%      121007.750000
max      124821.000000
Name: SEQN, dtype: float64

There are a total of 12634 unique SEQN in the whole dataset. So, we will randomly select a number between 109263 and 124821.

In [96]:
random.seed()
sample_id = random.sample(list(range(109263 ,124822)), 1)
sample_id

[120318]

In [81]:
merged_df.loc[merged_df['SEQN'] == 110718]

SEQN  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  TOT1FAT  \
3961  110718.0    1651.0     48.68    157.01     18.21      11.3     91.9   

      TOT1ATOC  TOT1SODI  TOT2KCAL  TOT2PROT  TOT2CARB  TOT2SUGR  TOT2FIBE  \
3961       8.6    3153.0       NaN       NaN       NaN       NaN       NaN   

      TOT2FAT  TOT2ATOC  TOT2SODI  
3961      NaN       NaN       NaN

In [82]:
merged_df.loc[merged_df['SEQN'] == 110718].T

3961
SEQN      110718.00
TOT1KCAL    1651.00
TOT1PROT      48.68
TOT1CARB     157.01
TOT1SUGR      18.21
TOT1FIBE      11.30
TOT1FAT       91.90
TOT1ATOC       8.60
TOT1SODI    3153.00
TOT2KCAL        NaN
TOT2PROT        NaN
TOT2CARB        NaN
TOT2SUGR        NaN
TOT2FIBE        NaN
TOT2FAT         NaN
TOT2ATOC        NaN
TOT2SODI        NaN

In [121]:
merged_df = pd.merge( merged_df, df_demo, on='SEQN', how='left')
merged_df

SEQN  DR1DAY  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  \
0      118027.0     6.0     771.0     41.45     89.37     28.08       8.4   
1      118039.0     5.0     849.0     21.17     80.91     30.36       9.6   
2      118040.0     3.0    3605.0    119.29    470.53    225.58      15.5   
3      118044.0     1.0    1864.0     68.89    128.94     40.15      13.4   
4      118046.0     6.0    1536.0     46.06    162.79     62.28      13.0   
...         ...     ...       ...       ...       ...       ...       ...   
12629  114946.0     6.0     760.0     18.07    131.07     81.21       7.9   
12630  115964.0     6.0    1264.0     35.36    163.69     81.67       9.4   
12631  117635.0     5.0    1027.0     30.75    148.87    142.92       0.1   
12632  117012.0     NaN       NaN       NaN       NaN       NaN       NaN   
12633  110939.0     NaN       NaN       NaN       NaN       NaN       NaN   

       TOT1FAT  TOT1ATOC  TOT1SODI  ...  FIAINTRP  MIALANG  MIAPROXY  \
0        29.39      2.12    1411.0  ...       2.0      NaN       NaN   
1        23.75     10.46     817.0  ...       2.0      1.0       2.0   
2       141.74     12.15    5932.0  ...       2.0      1.0       2.0   
3       119.34     10.03    4153.0  ...       2.0      1.0       2.0   
4        78.80      5.20    2521.0  ...       2.0      1.0       2.0   
...        ...       ...       ...  ...       ...      ...       ...   
12629    19.80      2.53     947.0  ...       2.0      NaN       NaN   
12630    54.41      5.64    1337.0  ...       NaN      NaN       NaN   
12631    37.51      1.36    1357.0  ...       2.0      1.0       2.0   
12632      NaN       NaN       NaN  ...       2.0      1.0       2.0   
12633      NaN       NaN       NaN  ...       2.0      1.0       2.0   

       MIAINTRP  AIALANGA      WTINTPRP      WTMECPRP  SDMVPSU  SDMVSTRA  \
0           NaN       NaN   1017.784203   1116.245058      2.0     151.0   
1           2.0       1.0  19656.879837  20483.507748      2.0     151.0   
2           2.0       1.0  10285.487702  11219.002476      2.0     166.0   
3           2.0       1.0  41308.946534  45691.636856      1.0     149.0   
4           2.0       1.0  32825.483325  36183.290773      1.0     163.0   
...         ...       ...           ...           ...      ...       ...   
12629       NaN       NaN  18433.564829  20724.099073      1.0     156.0   
12630       NaN       NaN  10004.607989  10031.782598      2.0     168.0   
12631       2.0       1.0  10029.002803  10341.065883      1.0     172.0   
12632       2.0       NaN   3472.828165   3772.691457      1.0     150.0   
12633       2.0       1.0  17248.628210  17989.886611      1.0     162.0   

           INDFMPIR  
0      4.900000e-01  
1      2.260000e+00  
2      2.800000e-01  
3      5.000000e+00  
4      1.610000e+00  
...             ...  
12629  3.910000e+00  
12630           NaN  
12631  1.200000e+00  
12632  7.000000e-01  
12633  5.397605e-79  

[12634 rows x 47 columns]

In [122]:
merged_df_1 = pd.merge(merged_df, df_BMX, how='left', on='SEQN')

In [126]:
merged_df_1

SEQN  DR1DAY  TOT1KCAL  TOT1PROT  TOT1CARB  TOT1SUGR  TOT1FIBE  \
0      118027.0     6.0     771.0     41.45     89.37     28.08       8.4   
1      118039.0     5.0     849.0     21.17     80.91     30.36       9.6   
2      118040.0     3.0    3605.0    119.29    470.53    225.58      15.5   
3      118044.0     1.0    1864.0     68.89    128.94     40.15      13.4   
4      118046.0     6.0    1536.0     46.06    162.79     62.28      13.0   
...         ...     ...       ...       ...       ...       ...       ...   
12629  114946.0     6.0     760.0     18.07    131.07     81.21       7.9   
12630  115964.0     6.0    1264.0     35.36    163.69     81.67       9.4   
12631  117635.0     5.0    1027.0     30.75    148.87    142.92       0.1   
12632  117012.0     NaN       NaN       NaN       NaN       NaN       NaN   
12633  110939.0     NaN       NaN       NaN       NaN       NaN       NaN   

       TOT1FAT  TOT1ATOC  TOT1SODI  ...  BMXLEG  BMILEG  BMXARML  BMIARML  \
0        29.39      2.12    1411.0  ...     NaN     NaN     19.5      NaN   
1        23.75     10.46     817.0  ...    37.6     NaN     35.5      NaN   
2       141.74     12.15    5932.0  ...    36.5     NaN     38.2      NaN   
3       119.34     10.03    4153.0  ...    43.7     NaN     42.1      NaN   
4        78.80      5.20    2521.0  ...     NaN     1.0      NaN      1.0   
...        ...       ...       ...  ...     ...     ...      ...      ...   
12629    19.80      2.53     947.0  ...     NaN     NaN     19.5      NaN   
12630    54.41      5.64    1337.0  ...     NaN     NaN     24.7      NaN   
12631    37.51      1.36    1357.0  ...    40.3     NaN     37.6      NaN   
12632      NaN       NaN       NaN  ...    28.5     NaN     26.0      NaN   
12633      NaN       NaN       NaN  ...    39.0     NaN     33.5      NaN   

       BMXARMC  BMIARMC  BMXWAIST  BMIWAIST  BMXHIP  BMIHIP  
0         16.5      NaN      45.2       NaN     NaN     NaN  
1         27.0      NaN      87.0       NaN   100.2     NaN  
2         33.0      NaN      96.6       NaN    99.7     NaN  
3         34.2      NaN      94.1       NaN   105.0     NaN  
4          NaN      1.0       NaN       1.0     NaN     1.0  
...        ...      ...       ...       ...     ...     ...  
12629     15.1      NaN      45.0       NaN     NaN     NaN  
12630     17.7      NaN      51.1       NaN     NaN     NaN  
12631     24.9      NaN      75.1       NaN    88.3     NaN  
12632     18.6      NaN      52.0       NaN     NaN     NaN  
12633     24.5      NaN      66.4       NaN    84.6     NaN  

[12634 rows x 68 columns]

In [127]:
merged_df_1.columns

Index(['SEQN', 'DR1DAY', 'TOT1KCAL', 'TOT1PROT', 'TOT1CARB', 'TOT1SUGR',
       'TOT1FIBE', 'TOT1FAT', 'TOT1ATOC', 'TOT1SODI', 'DR2DAY', 'TOT2KCAL',
       'TOT2PROT', 'TOT2CARB', 'TOT2SUGR', 'TOT2FIBE', 'TOT2FAT', 'TOT2ATOC',
       'TOT2SODI', 'SDDSRVYR', 'RIDSTATR', 'RIAGENDR', 'RIDAGEYR', 'RIDAGEMN',
       'RIDRETH1', 'RIDRETH3', 'RIDEXMON', 'DMDBORN4', 'DMDYRUSZ', 'DMDEDUC2',
       'DMDMARTZ', 'RIDEXPRG', 'SIALANG', 'SIAPROXY', 'SIAINTRP', 'FIALANG',
       'FIAPROXY', 'FIAINTRP', 'MIALANG', 'MIAPROXY', 'MIAINTRP', 'AIALANGA',
       'WTINTPRP', 'WTMECPRP', 'SDMVPSU', 'SDMVSTRA', 'INDFMPIR', 'BMDSTATS',
       'BMXWT', 'BMIWT', 'BMXRECUM', 'BMIRECUM', 'BMXHEAD', 'BMIHEAD', 'BMXHT',
       'BMIHT', 'BMXBMI', 'BMDBMIC', 'BMXLEG', 'BMILEG', 'BMXARML', 'BMIARML',
       'BMXARMC', 'BMIARMC', 'BMXWAIST', 'BMIWAIST', 'BMXHIP', 'BMIHIP'],
      dtype='object')

In [129]:
df = merged_df_1[['SEQN', 'DR1DAY', 'TOT1KCAL', 'TOT1PROT', 'TOT1CARB', 'TOT1SUGR', 'TOT1FIBE',
       'TOT1FAT', 'TOT1ATOC', 'TOT1SODI', 'DR2DAY', 'TOT2KCAL', 'TOT2PROT', 'TOT2CARB',
       'TOT2SUGR', 'TOT2FIBE', 'TOT2FAT', 'TOT2ATOC', 'TOT2SODI',
       'SIAPROXY', 'RIDAGEYR', 'RIDAGEMN', 'RIAGENDR', 'RIDRETH1', 'RIDRETH3', 'DMDMARTZ', 'DMDEDUC2', 'INDFMPIR', 'RIDEXPRG', 'DMDYRUSZ', 'SIALANG',
       'BMXBMI', 'BMDBMIC', 'BMXWT', 'BMXHT', 'BMXRECUM', 'BMXWAIST', 'BMXHIP']]

In [131]:
df[['SEQN', 'SIAPROXY', 'RIDAGEYR', 'RIDAGEMN', 'RIAGENDR', 'RIDRETH1', 'RIDRETH3', 'DMDMARTZ', 'DMDEDUC2', 'INDFMPIR', 'RIDEXPRG', 'DMDYRUSZ', 'SIALANG',
       'BMXBMI', 'BMDBMIC', 'BMXWT', 'BMXHT', 'BMXRECUM', 'BMXWAIST', 'BMXHIP']].loc[df['SEQN'] == 120318].T

3348
SEQN      120318.0
DR1DAY         7.0
SIAPROXY       1.0
RIDAGEYR      13.0
RIDAGEMN       NaN
RIAGENDR       2.0
RIDRETH1       1.0
RIDRETH3       1.0
DMDMARTZ       NaN
DMDEDUC2       NaN
INDFMPIR       NaN
RIDEXPRG       NaN
DMDYRUSZ       NaN
SIALANG        2.0
BMXBMI        27.1
BMDBMIC        4.0
BMXWT         74.7
BMXHT        165.9
BMXRECUM       NaN
BMXWAIST      95.3
BMXHIP       101.9

In [134]:
df[['SEQN', 'DR1DAY','TOT1KCAL', 'TOT1PROT', 'TOT1CARB', 'TOT1SUGR', 'TOT1FIBE','TOT1FAT', 'TOT1ATOC', 'TOT1SODI', 'DR2DAY', 'TOT2KCAL', 'TOT2PROT', 
    'TOT2CARB','TOT2SUGR', 'TOT2FIBE', 'TOT2FAT', 'TOT2ATOC', 'TOT2SODI']].loc[df['SEQN'] == 120318].T

3348
SEQN      120318.00
DR1DAY         7.00
TOT1KCAL    1023.00
TOT1PROT      24.77
TOT1CARB     162.78
TOT1SUGR      81.42
TOT1FIBE       9.40
TOT1FAT       30.43
TOT1ATOC       4.07
TOT1SODI    1557.00
DR2DAY         3.00
TOT2KCAL     716.00
TOT2PROT      34.60
TOT2CARB     100.08
TOT2SUGR      18.38
TOT2FIBE       5.20
TOT2FAT       20.18
TOT2ATOC       3.97
TOT2SODI    1214.00

In [140]:
df_dietary_1.columns

Index(['SEQN', 'WTDRD1PP', 'WTDR2DPP', 'DR1ILINE', 'DR1DRSTZ', 'DR1EXMER',
       'DRABF', 'DRDINT', 'DR1DBIH', 'DR1DAY', 'DR1LANG', 'DR1CCMNM',
       'DR1CCMTX', 'DR1_020', 'DR1_030Z', 'DR1FS', 'DR1_040Z', 'DR1IFDCD',
       'DR1IGRMS', 'DR1IKCAL', 'DR1IPROT', 'DR1ICARB', 'DR1ISUGR', 'DR1IFIBE',
       'DR1ITFAT', 'DR1ISFAT', 'DR1IMFAT', 'DR1IPFAT', 'DR1ICHOL', 'DR1IATOC',
       'DR1IATOA', 'DR1IRET', 'DR1IVARA', 'DR1IACAR', 'DR1IBCAR', 'DR1ICRYP',
       'DR1ILYCO', 'DR1ILZ', 'DR1IVB1', 'DR1IVB2', 'DR1INIAC', 'DR1IVB6',
       'DR1IFOLA', 'DR1IFA', 'DR1IFF', 'DR1IFDFE', 'DR1ICHL', 'DR1IVB12',
       'DR1IB12A', 'DR1IVC', 'DR1IVD', 'DR1IVK', 'DR1ICALC', 'DR1IPHOS',
       'DR1IMAGN', 'DR1IIRON', 'DR1IZINC', 'DR1ICOPP', 'DR1ISODI', 'DR1IPOTA',
       'DR1ISELE', 'DR1ICAFF', 'DR1ITHEO', 'DR1IALCO', 'DR1IMOIS', 'DR1IS040',
       'DR1IS060', 'DR1IS080', 'DR1IS100', 'DR1IS120', 'DR1IS140', 'DR1IS160',
       'DR1IS180', 'DR1IM161', 'DR1IM181', 'DR1IM201', 'DR1IM221', 'DR1IP182',
       

In [147]:
df_1 = df_dietary_1[['SEQN', 'DR1ILINE', 'DR1DRSTZ',
       'DRABF', 'DRDINT', 'DR1DAY', 'DR1LANG', 'DR1CCMNM',
       'DR1CCMTX', 'DR1_020', 'DR1_030Z', 'DR1FS', 'DR1_040Z', 'DR1IFDCD',
       'DR1IGRMS', 'DR1IKCAL', 'DR1IPROT', 'DR1ICARB', 'DR1ISUGR', 'DR1IFIBE',
       'DR1ITFAT', 'DR1ISFAT', 'DR1IMFAT', 'DR1IPFAT', 'DR1ICHOL', 'DR1IATOC',
       'DR1IATOA', 'DR1IRET', 'DR1IVARA', 'DR1IACAR', 'DR1IBCAR', 'DR1ICRYP',
       'DR1ILYCO', 'DR1ILZ', 'DR1IVB1', 'DR1IVB2', 'DR1INIAC', 'DR1IVB6',
       'DR1IFOLA', 'DR1IFA', 'DR1IFF', 'DR1IFDFE', 'DR1ICHL', 'DR1IVB12',
       'DR1IB12A', 'DR1IVC', 'DR1IVD', 'DR1IVK', 'DR1ICALC', 'DR1IPHOS',
       'DR1IMAGN', 'DR1IIRON', 'DR1IZINC', 'DR1ICOPP', 'DR1ISODI', 'DR1IPOTA',
       'DR1ISELE', 'DR1ICAFF', 'DR1ITHEO', 'DR1IALCO', 'DR1IMOIS']].loc[df_dietary_1['SEQN']==120318]

In [148]:
df_1.to_csv('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/Dietary_data_1_120318.csv')

In [150]:
df_dietary_2.columns

Index(['SEQN', 'WTDRD1PP', 'WTDR2DPP', 'DR2ILINE', 'DR2DRSTZ', 'DR2EXMER',
       'DRABF', 'DRDINT', 'DR2DBIH', 'DR2DAY', 'DR2LANG', 'DR2CCMNM',
       'DR2CCMTX', 'DR2_020', 'DR2_030Z', 'DR2FS', 'DR2_040Z', 'DR2IFDCD',
       'DR2IGRMS', 'DR2IKCAL', 'DR2IPROT', 'DR2ICARB', 'DR2ISUGR', 'DR2IFIBE',
       'DR2ITFAT', 'DR2ISFAT', 'DR2IMFAT', 'DR2IPFAT', 'DR2ICHOL', 'DR2IATOC',
       'DR2IATOA', 'DR2IRET', 'DR2IVARA', 'DR2IACAR', 'DR2IBCAR', 'DR2ICRYP',
       'DR2ILYCO', 'DR2ILZ', 'DR2IVB1', 'DR2IVB2', 'DR2INIAC', 'DR2IVB6',
       'DR2IFOLA', 'DR2IFA', 'DR2IFF', 'DR2IFDFE', 'DR2ICHL', 'DR2IVB12',
       'DR2IB12A', 'DR2IVC', 'DR2IVD', 'DR2IVK', 'DR2ICALC', 'DR2IPHOS',
       'DR2IMAGN', 'DR2IIRON', 'DR2IZINC', 'DR2ICOPP', 'DR2ISODI', 'DR2IPOTA',
       'DR2ISELE', 'DR2ICAFF', 'DR2ITHEO', 'DR2IALCO', 'DR2IMOIS', 'DR2IS040',
       'DR2IS060', 'DR2IS080', 'DR2IS100', 'DR2IS120', 'DR2IS140', 'DR2IS160',
       'DR2IS180', 'DR2IM161', 'DR2IM181', 'DR2IM201', 'DR2IM221', 'DR2IP182',
       

In [151]:
df_2 = df_dietary_2[['SEQN',  'DR2ILINE', 'DR2DRSTZ', 
       'DRABF', 'DRDINT', 'DR2DAY', 'DR2LANG', 'DR2CCMNM',
       'DR2CCMTX', 'DR2_020', 'DR2_030Z', 'DR2FS', 'DR2_040Z', 'DR2IFDCD',
       'DR2IGRMS', 'DR2IKCAL', 'DR2IPROT', 'DR2ICARB', 'DR2ISUGR', 'DR2IFIBE',
       'DR2ITFAT', 'DR2ISFAT', 'DR2IMFAT', 'DR2IPFAT', 'DR2ICHOL', 'DR2IATOC',
       'DR2IATOA', 'DR2IRET', 'DR2IVARA', 'DR2IACAR', 'DR2IBCAR', 'DR2ICRYP',
       'DR2ILYCO', 'DR2ILZ', 'DR2IVB1', 'DR2IVB2', 'DR2INIAC', 'DR2IVB6',
       'DR2IFOLA', 'DR2IFA', 'DR2IFF', 'DR2IFDFE', 'DR2ICHL', 'DR2IVB12',
       'DR2IB12A', 'DR2IVC', 'DR2IVD', 'DR2IVK', 'DR2ICALC', 'DR2IPHOS',
       'DR2IMAGN', 'DR2IIRON', 'DR2IZINC', 'DR2ICOPP', 'DR2ISODI', 'DR2IPOTA',
       'DR2ISELE', 'DR2ICAFF', 'DR2ITHEO', 'DR2IALCO', 'DR2IMOIS']].loc[df_dietary_2['SEQN']==120318]

In [152]:
df_2.to_csv('/Users/soumyadeepray/Library/Mobile Documents/com~apple~CloudDocs/Documents/Research_Project-Nutrition_Sciences/NHANES/data/Dietary_data_2_120318.csv')

In [68]:
df_total_nutrient_1['SEQN'].value_counts()

109263.0    1
119688.0    1
119676.0    1
119677.0    1
119678.0    1
           ..
114468.0    1
114469.0    1
114470.0    1
114471.0    1
124822.0    1
Name: SEQN, Length: 14300, dtype: int64

In [154]:
q_3 = db.sql('select count (distinct SEQN) from df_total_nutrient_1')
q_3

┌────────────────────────┐
│ count(DISTINCT "SEQN") │
│         int64          │
├────────────────────────┤
│                  14300 │
└────────────────────────┘

In [155]:
q_4 = db.sql('select count (distinct SEQN) from df_total_nutrient_2')
q_4

┌────────────────────────┐
│ count(DISTINCT "SEQN") │
│         int64          │
├────────────────────────┤
│                  14300 │
└────────────────────────┘

In [69]:
merged_total_nutrient_df = pd.merge(df_total_nutrient_1, df_total_nutrient_2, on='SEQN', how='outer')
merged_total_nutrient_df

SEQN    WTDRD1PP_x    WTDR2DPP_x  DR1DRSTZ  DR1EXMER  DRABF_x  \
0      109263.0  7.619484e+03  1.780807e+04       1.0      14.0      2.0   
1      109264.0  8.235896e+03  7.253762e+03       1.0      81.0      2.0   
2      109265.0  3.353508e+04  3.561201e+04       1.0      88.0      2.0   
3      109266.0  6.831068e+03  5.988204e+03       1.0      81.0      2.0   
4      109269.0  7.875707e+03  1.823193e+04       1.0      88.0      2.0   
...         ...           ...           ...       ...       ...      ...   
14295  124818.0  1.546208e+04  1.243740e+04       1.0      49.0      2.0   
14296  124819.0  4.091615e+03  4.100758e+03       1.0      14.0      2.0   
14297  124820.0  3.435783e+04  3.505186e+04       1.0      86.0      2.0   
14298  124821.0  3.043724e+03  5.397605e-79       1.0      81.0      2.0   
14299  124822.0  5.397605e-79           NaN       2.0      14.0      NaN   

       DRDINT_x       DR1DBIH  DR1DAY  DR1LANG  ...      DR2TP184  DR2TP204  \
0           2.0  4.000000e+00     6.0      1.0  ...  5.397605e-79     0.035   
1           2.0  5.000000e+00     6.0      1.0  ...  5.397605e-79     0.085   
2           2.0  1.900000e+01     4.0      1.0  ...  5.397605e-79     0.070   
3           2.0  4.000000e+00     7.0      1.0  ...  5.397605e-79     0.028   
4           2.0  9.000000e+00     1.0      1.0  ...  1.000000e-03     0.005   
...         ...           ...     ...      ...  ...           ...       ...   
14295       2.0  6.000000e+00     7.0      1.0  ...  4.000000e-03     0.075   
14296       2.0  2.200000e+01     4.0      1.0  ...  3.700000e-02     0.045   
14297       2.0  1.800000e+01     4.0      1.0  ...  2.000000e-03     0.078   
14298       1.0  5.397605e-79     7.0      1.0  ...           NaN       NaN   
14299       NaN           NaN     7.0      2.0  ...           NaN       NaN   

           DR2TP205  DR2TP225      DR2TP226  DR2_300      DR2_320Z  \
0      2.000000e-03     0.006  5.397605e-79      2.0  3.000000e+02   
1      5.000000e-03     0.004  3.000000e-03      2.0  1.080000e+03   
2      2.000000e-03     0.014  7.000000e-03      2.0  4.800000e+02   
3      6.000000e-03     0.013  2.000000e-03      2.0  4.057800e+03   
4      2.000000e-03     0.001  5.397605e-79      3.0  1.800000e+02   
...             ...       ...           ...      ...           ...   
14295  1.400000e-02     0.025  2.000000e-03      2.0  5.397605e-79   
14296  5.397605e-79     0.003  1.000000e-03      2.0  5.397605e-79   
14297  5.000000e-03     0.011  6.000000e-03      2.0  1.200000e+02   
14298           NaN       NaN           NaN      NaN           NaN   
14299           NaN       NaN           NaN      NaN           NaN   

           DR2_330Z      DR2BWATZ  DR2TWSZ  
0      3.000000e+02  5.397605e-79      1.0  
1      1.080000e+03  5.397605e-79      1.0  
2      4.800000e+02  5.397605e-79     91.0  
3      4.057800e+03  5.397605e-79      1.0  
4      5.397605e-79  1.800000e+02      1.0  
...             ...           ...      ...  
14295  5.397605e-79  5.397605e-79      4.0  
14296  5.397605e-79  5.397605e-79      1.0  
14297  1.200000e+02  5.397605e-79     91.0  
14298           NaN           NaN      NaN  
14299           NaN           NaN      NaN  

[14300 rows x 252 columns]

In [ ]:
q_5 = db.sql('select SEQN, DR1DAY, sum(DR1IKCAL) TOT1KCAL , sum(DR1IPROT) TOT1PROT, sum(DR1ICARB) TOT1CARB, sum(DR1ISUGR) TOT1SUGR, sum(DR1IFIBE) TOT1FIBE, sum(DR1ITFAT) TOT1FAT, sum(DR1IATOC) TOT1ATOC, sum(DR1ISODI) TOT1SODI from df_dietary_1 group by SEQN, DR1DAY').df()
q_5

In [71]:
merged_total_nutrient_df.loc[merged_total_nutrient_df['SEQN']==124822.0].T

14299
SEQN        1.248220e+05
WTDRD1PP_x  5.397605e-79
WTDR2DPP_x           NaN
DR1DRSTZ    2.000000e+00
DR1EXMER    1.400000e+01
...                  ...
DR2_300              NaN
DR2_320Z             NaN
DR2_330Z             NaN
DR2BWATZ             NaN
DR2TWSZ              NaN

[252 rows x 1 columns]